sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+   * @param {int}                [options.skip]                            The number of topics to skip.
+   * @return {Promise<HttpOperationResponse}
+   */
+  async listRules(
+    topicPath: string,
+    subscriptionPath: string,
+    options?: any
+  ): Promise<HttpOperationResponse> {
+    const fullPath =
+      ServiceBusAtomManagementClient.getSubscriptionPath(topicPath, subscriptionPath) + "/Rules/";
+    return this._listResources(fullPath, options, this.ruleResourceSerializer);
+  }
+
+  /**
+   * Formats a queue path to point towards its dead letter queue.
+   *
+   * @param {string}             queuePath                A string object that represents the name of the queue whose dead letter path you want.
+   * @return {string}
+   * The path to the queue's dead letter queue
+   */
+  formatDeadLetterPath(queuePath: any): any {
+    return `${queuePath}/$DeadLetterQueue`;
+  }
+
+  /**
+   * Creates a resource.
+   *
+   * @param {string}             path                                                The resource path.
+   * @param {object}             requestBody                                         The resource handler.
+   * @param {boolean}            isUpdate                                            `isUpdate` flag indicates whether the `PUT` request
+   *                                                                                 must attempt to update the resource or to create new one.
+   * @param {ResourceSerializer} serializer                                          The XML serializer to use.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _createResource(
+    path: string,
+    requestBody: any,
+    isUpdate: boolean = false,
+    serializer: ResourceSerializer
+  ): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
+
+    webResource.method = "PUT";
+    webResource.url = this.getUrl(path);
+    webResource.body = JSON.stringify(requestBody);
+    if (isUpdate) {
+      webResource.headers.set("If-Match", "*");
+    }
+    webResource.headers.set("content-type", "application/atom+xml;type=entry;charset=utf-8");
+    webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
+
+    const atomXmlOperationSpec: AtomXmlOperationSpec = {
+      serializer: serializer,
+      shouldParseResponse: false
+    };
+    webResource.atomXmlOperationSpec = atomXmlOperationSpec;
+
+    return this.sendRequest(webResource);
+  }
+
+  /**
+   * Gets a resource.
+   *
+   * @param {string}             path                                                The resource path.
+   * @param {ResourceSerializer} serializer                                          The XML serializer to use.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _getResource(
+    path: string,
+    serializer: ResourceSerializer
+  ): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
```

Comment:
Nit: prefer `const` over `let` if you only assign once, applicable anywhere in the code base.